test/goose/brokers/redis/batch_test.clj

```diff
@@ -58,15 +64,23 @@
 
       (redis-batch/enqueue tu/redis-conn batch)
 
-      (is (= (redis-cmds/parse-map tu/redis-conn batch-state-key) batch-state))
+      (is (= (->> (redis-cmds/parse-map tu/redis-conn batch-state-key)
+               (reduce (fn [m [k v]] (assoc m k (str v))) {}))
+             batch-state))
       (is (= (redis-cmds/set-members tu/redis-conn enqueued-job-set) job-ids)))))
 
 ;;;; ======= TEST: Batch state update middleware ===========
 
+(defn redis-set-batch-state [conn batch]
+  (let [batch-state-key (d/prefix-batch batch-id)
+        batch-state (redis-batch/batch-state batch)]
+    (redis-cmds/hmset* conn batch-state-key batch-state)))
+
 (deftest enqueued-to-successful
   (testing "Job state is transitioned from enqueued -> successful on successful job execution"
     (redis-cmds/add-to-set (:redis-conn opts) enqueued-job-set job-id)
-    ((redis-batch/update-state (fn [_opts _job] "Function executed"))
+    (redis-set-batch-state (:redis-conn opts) batch)
+    ((redis-batch/wrap-state-update (fn [_opts _job] "Function executed"))
```

Comment:
comment: this double parentheses was confusing for a second, maybe add a comment saying, calling the middleware for batch state-update

Reply:
Added a comment in all unit tests where we're calling the middleware.